docs/go_api.md

```diff
@@ -0,0 +1,342 @@
+
+Our [demo Rust application](https://github.com/second-state/WasmEdge-go-examples/tree/master/go_ReadFile/rust_readfile) reads from a file. Note that there is no need for #{wasm_bindgen] here, as the WebAssembly program’s input and output data are now passed by the STDIN and STDOUT.
+
+
+```rust
+use std::env;
+use std::fs::File;
+use std::io::{self, BufRead};
+
+fn main() {
+    // Get the argv.
+    let args: Vec<String> = env::args().collect();
+    if args.len() <= 1 {
+        println!("Rust: ERROR - No input file name.");
+        return;
+    }
+
+    // Open the file.
+    println!("Rust: Opening input file \"{}\"...", args[1]);
+    let file = match File::open(&args[1]) {
+        Err(why) => {
+            println!("Rust: ERROR - Open file \"{}\" failed: {}", args[1], why);
+            return;
+        },
+        Ok(file) => file,
+    };
+
+    // Read lines.
+    let reader = io::BufReader::new(file);
+    let mut texts:Vec<String> = Vec::new();
+    for line in reader.lines() {
+        if let Ok(text) = line {
+            texts.push(text);
+        }
+    }
+    println!("Rust: Read input file \"{}\" succeeded.", args[1]);
+
+    // Get stdin to print lines.
+    println!("Rust: Please input the line number to print the line of file.");
+    let stdin = io::stdin();
+    for line in stdin.lock().lines() {
+        let input = line.unwrap();
+        match input.parse::<usize>() {
+            Ok(n) => if n > 0 && n <= texts.len() {
+                println!("{}", texts[n - 1]);
+            } else {
+                println!("Rust: ERROR - Line \"{}\" is out of range.", n);
+            },
+            Err(e) => println!("Rust: ERROR - Input \"{}\" is not an integer: {}", input, e),
+        }
+    }
+    println!("Rust: Process end.");
+}
+```
+
+Use the rustwasmc tool to compile the application into WebAssembly.
+
+```bash
+$ cd rust_readfile
+$ rustwasmc build
+# The output file will be pkg/rust_readfile.wasm
+```
+
+The Golang source code to run the WebAssembly function in WasmEdge is as follows.
+
+```go
+package main
+
+import (
+    "os"
+    "github.com/second-state/WasmEdge-go/wasmedge"
+)
+
+func main() {
+    wasmedge.SetLogErrorLevel()
+
+    var conf = wasmedge.NewConfigure(wasmedge.REFERENCE_TYPES)
+    conf.AddConfig(wasmedge.WASI)
+    var vm = wasmedge.NewVMWithConfig(conf)
+    var wasi = vm.GetImportObject(wasmedge.WASI)
+    wasi.InitWasi(
+        os.Args[1:],     /// The args
+        os.Environ(),    /// The envs
+        []string{".:."}, /// The mapping directories
+        []string{},      /// The preopens will be empty
+    )
+
+    /// Instantiate wasm. _start refers to the main() function
+    vm.RunWasmFile(os.Args[1], "_start")
+
+    vm.Delete()
+    conf.Delete()
+}
+```
+
+Next, let's build the Golang application with the WasmEdge Golang SDK.
+
+```bash
+$ go get -u github.com/second-state/WasmEdge-go
```

Comment:
It's better to add the version: `go get github.com/second-state/WasmEdge-go/wasmedge@v0.9.0-rc3`, or the `v0.8.2` will be get.